### ISSUE TITLE
ca65: Offsets for .tags within structs (feature request)

### ISSUE THREAD

**Issue by inexorabletash** [cc65/cc65#665]
Using `.tag` within a `.struct` block doesn't create a "nested scope" for the offsets like nested `.struct` sections do. But nested structs don't behave particularly intuitively either:
```
.struct Point
    xcoord .word
    ycoord .word
.endstruct

.struct Rect
    pt1 .tag Point
    .struct pt2
        xcoord .word
        ycoord .word
    .endstruct
.endstruct

offset1: .byte Rect::pt1::ycoord       ; Error - would be nice if this evaluated to 2.
offset2: .byte Rect::pt2::ycoord       ; Compiles... but yields 2 not 6, alas. Can't fix, though.
```

This is mentioned in the docs as: _"adding an embedded structure to another structure with .TAG will not make this structure accessible by using the '::' operator."_

**Comment by oliverschmidt** [comment]
Given that you write "This is mentioned in the docs [...]" I presume this issue isn't a question, right? So is it a feature request?

**Comment by greg-king5** [comment]
> But, nested structs don't behave particularly intuitively, either.

Yep!  We have wanted C-style .structs ever since `.struct` was put into ca65.  The namespace operator (`::`) works on scopes.  We want a "member operator" (`.`) that works on labels.
```asm
.struct Rect
    pt1 .tag Point
pt2 .struct pt2
        xcoord .word
        ycoord .word
    .endstruct
.endstruct

box: .tag Rect
```
The first `pt2` is a label; and, the second `pt2` is a scope.  `Rect::pt2` refers to the label; but, `Rect::pt2::` refers to the scope.  Therefore, `Rect::pt2::ycoord` is an offset within `.struct pt2`.

`box.pt2.ycoord` would give us the address that we want -- if that feature were to be implemented.

**Comment by oliverschmidt** [comment]
> says so in the title

Ooops.

> I don't have permission to assign labels

I agree that it would make sense to allow the original author of an issue to assign label once - but GitHub doesn't...
